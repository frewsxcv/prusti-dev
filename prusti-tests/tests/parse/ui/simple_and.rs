/// Tests that parser handles spans correctly.

use prusti_contracts::*;

#[requires(12345)]
pub fn test_1(x: i32) {}

#[requires     (12345)]
pub fn test_2(x: i32) {}

#[

    requires
    
        (

    12345) ]
pub fn test_3(x: i32) {}

#[requires (

    12345)
    ]
pub fn test_4(x: i32) {}

#[requires(true && 12345)]
pub fn test_5(x: i32) {}

#[requires(true &&

    12345  
    
        &&  true && 
            false &&
        
            456)]
pub fn test_6(x: i32) {}

fn main() {}
