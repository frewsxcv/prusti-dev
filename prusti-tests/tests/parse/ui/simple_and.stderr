error[E0308]: mismatched types
 --> $DIR/simple_and.rs:5:12
  |
5 | #[requires(12345)]
  |            ^^^^^ expected `bool`, found integer

error[E0308]: mismatched types
 --> $DIR/simple_and.rs:8:17
  |
8 | #[requires     (12345)]
  |                 ^^^^^ expected `bool`, found integer

error[E0308]: mismatched types
  --> $DIR/simple_and.rs:17:5
   |
17 |     12345) ]
   |     ^^^^^ expected `bool`, found integer

error[E0308]: mismatched types
  --> $DIR/simple_and.rs:22:5
   |
22 |     12345)
   |     ^^^^^ expected `bool`, found integer

error[E0308]: mismatched types
  --> $DIR/simple_and.rs:26:20
   |
26 | #[requires(true && 12345)]
   |                    ^^^^^ expected `bool`, found integer

error[E0308]: mismatched types
  --> $DIR/simple_and.rs:31:5
   |
31 |     12345  
   |     ^^^^^ expected `bool`, found integer

error[E0308]: mismatched types
  --> $DIR/simple_and.rs:36:13
   |
36 |             456)]
   |             ^^^ expected `bool`, found integer

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0308`.
